### ISSUE TITLE
Don't expire old refreshToken when issuing a new one

### ISSUE THREAD

**Issue by bkonetzny** [bshaffer/oauth2-server-php#529]
I tried to create an OAuth2 protected IFTTT channel with this library. Everything works fine except one IFTTT authentication test which has the following title: "Ensure older refresh token does not expire immediately". IFTTT expects a valid /oauth2/token response for an already expired refresh token.

I searched the code and found `RefreshToken->createAccessToken`, when configured with `$this->config['always_issue_new_refresh_token'] = true` will issue a new refresh token when getting a new access token, but also invalidate the old refresh token. Comment in this method says this is because http://tools.ietf.org/html/rfc6749#section-6. I looked up this RFC and it says the client MUST discard the old refresh token, not the server:
> The authorization server MAY issue a new refresh token, in which case the client MUST discard the old refresh token and replace it with the new refresh token. The authorization server MAY revoke the old refresh token after issuing a new refresh token to the client.

Would it be possible to make unsetRefreshToken also an configurable option like this?
````
if ($issueNewRefreshToken && $this->config['always_unset_old_refresh_token']) {
  $this->storage->unsetRefreshToken($this->refreshToken['refresh_token']);
}
````

Related: https://github.com/bshaffer/oauth2-server-php/issues/169 and https://github.com/bshaffer/oauth2-server-php/issues/226

**Comment by bshaffer** [comment]
I think that's a great idea, and I love the thorough research you have done. Thank you!

**Comment by Jacketbg** [comment]
Hey I'm having the same issue here, exactly with IFTTT integration. Otherwise great library, but this issue seems to have no work-around without hacking your code. I don't really want to to this, because we're using Composer to deploy to several prod servers.
Do you think this change will made it into master branch anytime soon? Thanks.

**Comment by bshaffer** [comment]
sure! I was hoping for a Pull request from the OP, which is always the fastest way to get a change in. But I can probably add this by the end of the week otherwise.

**Comment by Jacketbg** [comment]
I just made a Pull request in regards to this issue.

**Comment by bshaffer** [comment]
resolved in #571
